### PR TITLE
Add activity showcasing style cross fading

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add activity showcasing style cross fading.</changelog>`.

### Summary of changes

Add activity showcasing style cross fading. 
For easier use added utility class of `StyleSwitcher` to `:app` module (not part of SDK itself).

### User impact (optional)

Before:

https://user-images.githubusercontent.com/15800566/122414411-cce89680-cf8f-11eb-8a30-92e934b135b8.mp4

After:

https://user-images.githubusercontent.com/15800566/122414084-8c891880-cf8f-11eb-9cc4-332d1ff9101e.mp4

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            }
+          }
+        }
+      },
+      listOf(MapEvents.MAP_LOADED)
+    )
+    currentStyle = initialStyle
+  }
+
+  fun switchStyle(
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:53]: Could actually be an extension function on MapboxMap? Current style get be got through MapboxMap#getStyle API.

Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:53]: Assuming it's not part of SDK but some util - I like current approach more (especially taking into account that this class has a valid ctor).

Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:53]: yeah, for example activity I think it doesn't really matter, although we recommend users not keeping a reference of the style object, and always get it from the mapboxmap async. Do you think we can optimise it that way(remove the initial style and current style)?

File: app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  init {
+    mapboxMap.subscribe(
+      object : Observer() {
+        override fun notify(event: Event) {
+          when (event.type) {
+            // when map loaded, fade-in layers using original values
+            MapEvents.MAP_LOADED -> {
+              for (originalProperty in originalProperties) {
+                currentStyle.setStyleLayerProperty(
+                  originalProperty.key.first,
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:27]: can we assume the layer still exist?

Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:27]: Are there any objections why it should not exist? IMHO `MAP_LOADED` is called almost immediately after `STYLE_LOADED` so those properties should be in place.

Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:27]: But will add a check.

Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:27]: I mean users could modify the style after you constructed the StyleSwitcher, the previously added layer/properties may not exist any more.

**Review by pengdev — APPROVED** [review]

File: app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
+package com.mapbox.maps.testapp.utils
+
+import android.os.Handler
+import android.os.Looper
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.*
+
+class StyleSwitcher(
```
[the remark this answers is not in the document]
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:14]: fixed.

File: app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt (hunk 1 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:61]: Done, thanks!

File: app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                  Value(0.0), Value(0.0), Value(0.0), Value(0.0)
+                )
+              )
+            )
+          }
+        }
+      }
+      mainHandler.postDelayed(
+        {
+          addOnMapLoadedObserver()
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:94]: nit: do we need to unsubscribe previous observer?

Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/utils/StyleSwitcher.kt:94]: we do it after `MAP_LOADED` event is received.

**Review by pengdev — APPROVED** [review]
Nice work! LGTM

**Pull request merged**